scripts/autogen

```diff
@@ -1395,6 +1395,54 @@ def check_asm_register_aliases():
         check_asm_register_aliases_for_file(asm_file)
 
 
+def gen_asm_loop_labels_for_file(filename, dry_run=False):
+    """Checks that all labels in `filename` are prefixed according to the function defined in the file"""
+    status_update("asm-labels", filename)
+
+    with open(filename, "r") as f:
+        content = f.read()
+
+    # Find function symbol name
+    func_pattern = r"MLK_ASM_FN_SYMBOL\((.*)\)"
+    res = re.search(func_pattern, content)
+    if res is None:
+        raise Exception(f"Could not find function symbol in assembly file {filename}")
+    funcname = res.group(1)
+    lbl_prefix = funcname.replace("_asm", "") + "_"
+    lbl_pattern = r"^(\w+):"
+    rename_labels = []
+    for m in re.finditer(lbl_pattern, content, flags=re.M):
+        lbl = m.group(1)
+        if lbl.startswith(lbl_prefix) and "asm_" not in lbl:
+            continue
+        rename_labels.append(lbl)
+
+    # Replace further uses of renamed labels
+    for lbl in rename_labels:
+        # Fix up bad attempts to prefix
+        funcname_parts = funcname.split("_")
+        lbl_parts = lbl.split("_")
+        lbl_stripped = lbl
+        for i, p in enumerate(lbl_parts):
+            if p not in funcname_parts:
+                lbl_stripped = "_".join(lbl_parts[i:])
+                break
+        new_lbl = lbl_prefix + lbl_stripped
+        content = re.sub(rf"(^|\s){lbl}(:?)$", rf"\1{new_lbl}\2", content, flags=re.M)
+
+    update_file(filename, content, dry_run=dry_run)
```

Comment:
I don't see much value in auto-renaming those. We don't ever want that to happen as the inputs should already contain the namespace. I can only see this leading to confusion when adding further backend. Can we instead change this to a fail with a reasonable error "assembly label XXX is missing function-name prefix"? Or am I missing something here?

Reply:
I can do that, yes. It was mainly for me to systematically change the existing labels.

Reply:
I'll do this as a follow-up commit, so the history retains the adapting version.

Reply:
Adjusted